platform/darwin/src/MGLPolyline.mm

```diff
@@ -52,6 +53,27 @@ - (BOOL)isEqual:(id)other {
     return self == other || ([other isKindOfClass:[MGLPolyline class]] && [super isEqual:other]);
 }
 
+- (CLLocationCoordinate2D)coordinate {
+    mbgl::Polygon<double> polyline;
```

Comment:
This code is incorrectly treating a polyline as if it's a polygon. Pole of inaccessibility isn't the right tool to use for calculating the midpoint of a polyline. Instead, you would need to manually [determine the length of the polyline](https://github.com/Turfjs/turf/tree/master/packages/turf-line-distance) and [find the point half of that distance](https://github.com/Turfjs/turf/tree/master/packages/turf-along) from the start. We've already ported this functionality to Swift as part of [Geometry.swift](https://github.com/mapbox/mapbox-navigation-ios/blob/master/MapboxCoreNavigation/Geometry.swift). Feel free to take care of this in a separate PR.